test/unit/transform/epacems_test.py

```diff
@@ -37,3 +41,85 @@ def test_harmonize_eia_epa_orispl():
     )
     actual_df = epacems.harmonize_eia_epa_orispl(cems_test_df, crosswalk_test_df)
     pd.testing.assert_frame_equal(expected_df, actual_df, check_dtype=False)
+
+
+def test_epacamd_eia_subplant_ids():
+    """Ensure subplant_id gets applied appropriately to example plants."""
+    clean_epacamd_eia_test = pd.read_csv(
+        StringIO(
+            """
+plant_id_epa,emissions_unit_id_epa,generator_id_epa,plant_id_eia,boiler_id,generator_id
+1391,1A,1A,1391,1A,1A
+1391,2A,2A,1391,2A,2A
+1391,3A,3A,1391,3A,3A
+1391,4A,4A,1391,,4A
+1391,5A,5A,1391,,5A
+1,A,A,1,,A
+"""
+        )
+    )
+
+    emissions_unit_ids_epacems_test = pd.read_csv(
+        StringIO(
+            """
+plant_id_eia,emissions_unit_id_epa
+1391,1A
+1391,2A
+1391,3A
+1391,4A
+1391,5A
+1,A
+"""
+        )
+    )
+
+    boiler_generator_assn_eia860_test = pd.read_csv(
+        StringIO(
+            """
+plant_id_eia,generator_id,unit_id_pudl
+1391,1A,1
+1391,2A,1
+1391,3A,1
+1,A,1
+"""
+        )
+    )
+
+    generators_entity_eia_test = pd.read_csv(
+        StringIO(
+            """
+plant_id_eia,generator_id
+1391,1A
+1391,2A
+1391,3A
+1391,4A
+1391,5A
+1,A
+"""
+        )
+    )
+    epacamd_eia_subplant_ids_expected = pd.read_csv(
+        StringIO(
+            """
+plant_id_eia,plant_id_epa,unit_id_pudl,emissions_unit_id_epa,generator_id,subplant_id
+1391,1391,1,1A,1A,0
+1391,1391,1,2A,2A,0
+1391,1391,1,3A,3A,0
+1391,1391,,4A,4A,1
+1391,1391,,5A,5A,2
+1,1,1,A,A,0
+"""
```

Comment:
hey @grgmiller @aesharpe ... when writing test for this subplant id , austen suggested I use the actual plant_id_eia 1391 as an example. But you both probably know better than I about what the desired behavior for this plant? I saw that greg you were manually assigning all of these as 0. Should we keep this manual assign? or is this behavior outlined in this little unit test expected?

Reply:
I had been manually assigning all generators at 1391 to the same subplant due to the associations in the BGA table. So I think that the expected behavior is that all subplant IDs are the same, but I think that this won't be the case until the BGA table is incorporated into the subplant ID code.

Reply:
I think in this specific case, we actually assigned 4A and 5A to subplant 0 due to a reason not reflected in the BGA table. We noticed that the minimum generation reported in CEMS for subplant 0 was higher than the nameplate capacity for that subplant, which suggested that there was something strange with the subplant mapping (see https://github.com/singularity-energy/open-grid-emissions/pull/142#issuecomment-1186579260). However, we should probably remove this manual update in the pudl code since it is not verified. I think the mapping in this unit test is probably correct from the standpoint of what is reported in the EPA crosswalk and BGA table, although this mapping leads to strange results in the actual data (e.g. the generation for subplant 0 is greater than the nameplate capacity of subplant 0, and no generation is reported for subplant 1 or 2). I think this suggests an issue with the underlying data potentially.

Reply:
okay i added this manually assign everything to 0 step back in. I'm curious if you have a sense of if this is an upstream cems problem or something else. if you know! i know tracking these little guys down to really understand (plus keeping it in your head/in docs) can be challenging.